### DISCUSSION TITLE
How do I composite a cropped section of an image over top of another image?

### DISCUSSION THREAD

**Discussion by briankendall** [ImageMagick/ImageMagick#7563] [Help] [answered]
### ImageMagick version

7.1.1-34

### Operating system, version and so on

macOS 13.6.5

### Description

I've been pulling my hair out trying to figure out the correct syntax to composite a cropped section of an image over top of another image at a specific position. Ideally I would be able to use a single command to composite several cropped sections of an image, some of which may come from the same image. But I want to walk before I run.

Some of the commands I've tried so far are:

`magick section.png -crop 24x24+24+0 background.png -composite result.png`
`magick \( section.png -crop 24x24+24+0 \) background.png -composite result.png`
`magick \( section.png -crop 24x24+24+0 +repage \) background.png -composite result.png`

All of these just produces a 24x24 image containing a cropped part of background.png and without containing the portion of `section.png` that I'm trying to crop. I don't understand why the wrong image is being cropped -- shouldn't `--crop` apply to the image before it, especially with the parenthesis?

I've been doing tons of searches using different search engines trying to find information about how to do this, or any examples of a crop operation being combined with compositing, but I haven't found anything. It seems like ImageMagick's new syntax is very flexible and is capable of doing this, though what seems intuitive to me keeps ending up with errors or the wrong result, such as the above.

**Comment by fmw42** [comment, marked as answer]
Put the background image first and add +repage after the crop

`magick background \( section.png -crop 24x24+24+0 +repage \) -compose over -composite result.png`

or 

`magick \( section.png -crop 24x24+24+0 +repage \) background.png +swap -compose over -composite result.png`

The -compose over is optional, since that is the default for -composite.

**Reply by briankendall** [reply]
Yes, that got it, thank you. Just so that I may learn to fish rather than be given a fish, so to speak, was the issue simply that I was reversing the order of the images?

**Reply by fmw42** [reply]
Yes.  See https://imagemagick.org/Usage/compose/#compose

For novices, see

http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=9620
http://www.imagemagick.org/script/command-line-processing.php
https://imagemagick.org/script/command-line-options.php (and see the sitemap link at the bottom)
http://www.imagemagick.org/Usage/reference.html
http://www.imagemagick.org/Usage/
https://github.com/ImageMagick/usage-markdown
https://imagemagick.org/script/porting.php#cli
https://legacy.imagemagick.org/discourse-server/index.php
